### PR TITLE
Embedded Pg v1.0.1

### PR DESCRIPTION
**Pull request by rorp**
Allows to run Postgres tests on M1 Macs, requires docker installed on the developers desktop environment.

### CODE REVIEW

File: testkit/src/main/scala/org/bitcoins/testkit/fixtures/WalletDAOFixture.scala [hunk 1]
```diff
@@ -63,7 +63,17 @@ trait WalletDAOFixture extends BitcoinSFixture with EmbeddedPg {
...
+          _ <- config.schemaName match {
+            case Some(schemaName) =>
+              import config.dbConfig.profile.api._
+              config.database
+                .run(sqlu"""DROP SCHEMA IF EXISTS #$schemaName CASCADE""")
+                .map(_ => ())
+            case None => Future.unit
+          }
+          _ = config.migrate()
+        } yield daos
```
Comment (benthecarman) [inline testkit/src/main/scala/org/bitcoins/testkit/fixtures/WalletDAOFixture.scala]: I think you can replace this by instead calling `config.clean()` in `dropAll()`

**Comment by benthecarman** [comment]
Windows tests seem to still be failing, might need to install docker

**Comment by Christewart** [comment]
@rorp This doesn't work for me locally on baa2282a1a6b0b48eabde73901513c8d6f669268 

However it does leave a bunch of docker containers running locally.

Screenshot from 2022-08-04 14-00-46

**Comment by rorp** [comment]
> @rorp This doesn't work for me locally on [baa2282](https://github.com/bitcoin-s/bitcoin-s/commit/baa2282a1a6b0b48eabde73901513c8d6f669268)

Since it does work on OSX both Intel and ARM, and even in CI, the issue is definitely in your local environment :)

**Comment by Christewart** [comment]
@benthecarman Can you try this PR locally `PG_ENABLED=1`

**Comment by Christewart** [comment]
> > @rorp This doesn't work for me locally on [baa2282](https://github.com/bitcoin-s/bitcoin-s/commit/baa2282a1a6b0b48eabde73901513c8d6f669268)
> 
> Since it does work on OSX both Intel and ARM, and even in CI, the issue is definitely in your local environment :)


Screen Shot 2022-08-04 at 2 45 48 PM

**Comment by rorp** [comment]
Screenshots of command line apps are useless for debugging. One can't even copy and paste an error message or a stack trace. This one doesn't fit the error message. Consider post logs, since they are way more informative.

My guess that your computer is too slow. Can you try the CI settings: https://github.com/bitcoin-s/bitcoin-s/pull/4567/files#diff-d0439890bf9a6d9f19466a0b876808618930c79501e5b208f51927046afdf3f1R15 ?

**Comment by benthecarman** [comment]
> @benthecarman Can you try this PR locally `PG_ENABLED=1`

```
[info] org.bitcoins.wallet.BitcoindBlockPollingTest *** ABORTED *** (47 seconds, 916 milliseconds)
[info]   org.testcontainers.containers.ContainerLaunchException: Container startup failed
[info]   at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:336)
[info]   at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:317)
[info]   at com.opentable.db.postgres.embedded.EmbeddedPostgres.<init>(EmbeddedPostgres.java:99)
[info]   at com.opentable.db.postgres.embedded.EmbeddedPostgres$Builder.start(EmbeddedPostgres.java:353)
[info]   at org.bitcoins.testkit.EmbeddedPg.pg(EmbeddedPg.scala:31)
[info]   at org.bitcoins.testkit.EmbeddedPg.pg$(EmbeddedPg.scala:23)
```

**Comment by benthecarman** [comment]
We use pg embedded v1.0.1 in the tbc backend, this is our embeddedpg.scala

```scala
package com.thebitcoincompany.testkit

import com.opentable.db.postgres.embedded.EmbeddedPostgres
import com.thebitcoincompany.db.config.TBCGiftCardDbConfig
import com.typesafe.config.{Config, ConfigFactory}
import org.scalatest.{BeforeAndAfterAll, Suite}

trait EmbeddedPg extends BeforeAndAfterAll { this: Suite =>

  lazy val pg: EmbeddedPostgres = {
    EmbeddedPostgres
      .builder()
      .setServerConfig("max_connections", "25")
      .setServerConfig("shared_buffers", "1MB")
      .start()
  }

  def pgUrl(): String =
    pg.getJdbcUrl("postgres")

  override def afterAll(): Unit = {
    super.afterAll()
    val _ = pg.close()
    ()
  }

  /** Generates a Typesafe config with DBs set to memory
    * databases for the given project (or all, if no
    * project is given). This configuration can then be
    * given as a override to other configs.
    */
  def configWithEmbeddedDb(pgUrl: () => String): Config = {
    val confStr: String = {
      val projectString = TBCGiftCardDbConfig.moduleName
      val url = pgUrl()
      val parts = url.split(":")
      require(parts.size >= 3 && parts(0) == "jdbc",
              s"`$url` must be a valid JDBC URL")
      val str = parts(3)
      val endOfPortStr = str.indexOf('/')
      val (port, _) = str.splitAt(endOfPortStr)

      val poolName =
        s"bitcoin-s-$projectString-pool-${System.currentTimeMillis()}"

      s"""$projectString.profile = "slick.jdbc.PostgresProfile$$"
         | $projectString.db {
         |   name = postgres
         |   url = "$url"
         |   driver = "org.postgresql.Driver"
         |   user = "postgres"
         |   password = "postgres"
         |   poolName = "$poolName"
         |   host = "localhost"
         |   port = $port
         |   numThreads = 1
         |   keepAliveConnection = true
         | }""".stripMargin
    }

    val nestedConfStr = s"""
                           | bitcoin-s {
                           | $confStr
                           | }
                           |""".stripMargin
    ConfigFactory.parseString(nestedConfStr)
  }
}
```

**Comment by Christewart** [comment]
> Screenshots of command line apps are useless for debugging. One can't even copy and paste an error message or a stack trace. This one doesn't fit the error message. Consider post logs, since they are way more informative.

[pg-embedded-debug.txt](https://github.com/bitcoin-s/bitcoin-s/files/9262898/pg-embedded-debug.txt)

**Comment by Christewart** [comment]
As discussed on the call with @rorp , restarted by docker instance with `sudo service docker restart`. When docker booted up, i had no lingering postgres containers. I checked with `docker ps -a`. I then tried to run a test suite (`chainTest/test`) with `PG_ENABLED=1` and got the same behavior of test fails and docker containers not getting shutdown. This is on linux, but i've seen similar behavior on my mac.

```
CONTAINER ID   IMAGE                       COMMAND                  CREATED         STATUS         PORTS                                         NAMES
24fc60272002   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49177->5432/tcp, :::49177->5432/tcp   priceless_goodall
54ca3a5ec1ce   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49172->5432/tcp, :::49172->5432/tcp   competent_gagarin
a0e689deaca5   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49170->5432/tcp, :::49170->5432/tcp   heuristic_dubinsky
d384209c10ff   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49174->5432/tcp, :::49174->5432/tcp   focused_williamson
0e253124306c   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49168->5432/tcp, :::49168->5432/tcp   practical_roentgen
4404c31c4912   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49173->5432/tcp, :::49173->5432/tcp   sad_borg
d2ce1896e02f   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49175->5432/tcp, :::49175->5432/tcp   stoic_edison
5cef31c0dde2   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49171->5432/tcp, :::49171->5432/tcp   keen_kilby
73cb95e40ecf   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49169->5432/tcp, :::49169->5432/tcp   trusting_solomon
ec4208d435b4   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49176->5432/tcp, :::49176->5432/tcp   intelligent_goldstine
eea9218904ec   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49167->5432/tcp, :::49167->5432/tcp   modest_maxwell
05542ba3628c   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49166->5432/tcp, :::49166->5432/tcp   dazzling_blackburn
4b51eb3bdaa9   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49161->5432/tcp, :::49161->5432/tcp   great_lewin
c9f2d76d8ff6   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49165->5432/tcp, :::49165->5432/tcp   strange_turing
22940000befe   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49164->5432/tcp, :::49164->5432/tcp   optimistic_mcclintock
eeb61c432f67   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49155->5432/tcp, :::49155->5432/tcp   sleepy_jemison
9975739eaa16   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49159->5432/tcp, :::49159->5432/tcp   nostalgic_babbage
8227c76aadd5   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49158->5432/tcp, :::49158->5432/tcp   xenodochial_joliot
e1de975cd2de   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49160->5432/tcp, :::49160->5432/tcp   jovial_clarke
fa97bc46245b   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49154->5432/tcp, :::49154->5432/tcp   practical_tu
5f0ae37b2ed4   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49157->5432/tcp, :::49157->5432/tcp   naughty_tesla
0968b5dae39e   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49156->5432/tcp, :::49156->5432/tcp   mystifying_pare
272e777fee60   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49162->5432/tcp, :::49162->5432/tcp   inspiring_moore
2786d2c4cd1c   postgres:13-alpine          "docker-entrypoint.s…"   5 minutes ago   Up 5 minutes   0.0.0.0:49163->5432/tcp, :::49163->5432/tcp   youthful_aryabhata
8dc2b1c77d82   testcontainers/ryuk:0.3.3   "/app"                   6 minutes ago   Up 5 minutes   0.0.0.0:49153->8080/tcp, :::49153->8080/tcp   testcontainers-ryuk-ab4deb95-8d77-4912-a781-6fc783dd254c
```

**Pull request merged**